app/decorators/application_decorator.rb

```diff
@@ -1,5 +1,29 @@
+# frozen_string_literal: true
+
 class ApplicationDecorator < Draper::Decorator
   delegate_all
-  
+
   self.include_root_in_json = true
+
+  private
+
+  API_KEYS = %w[id name system_name].freeze
+
+  def parse_api(api_hash)
+    add_link(add_api_type(api_hash.slice(*API_KEYS)))
+  end
+
+  def add_link(api_hash)
+    api_hash[:link] = api_selector_api_link
```

Comment:
This `api_selector_api_link` needs to be defined in the `ApplicationDecorator`
Not obvious that your Application Interface requries it